### DISCUSSION TITLE
Laravel 8 Upgrade and general roadmap plans?

### DISCUSSION THREAD

**Discussion by lostdesign** [area17/twill#856] [General]
Hi there, seeing that tomorrow is L7 last day of bugfix support and having no further security support around march 2021, I'd like to ask what the [roadmap](https://spectrum.chat/twill/announcements/area17-staff-is-there-going-to-be-a-public-road-map~5b65518c-3424-40d7-9c69-6af377774c2d) is for upgrading to L8 and beyond.

Laravel adapted a rather quick release cycle, which twill seems to be behind judging based on the last upgrade to version 7.
I do know this is an open source project and I am not trying to say that there are any obligations from Area17's side, this is mainly to vet a decision going further with twill in general.

This also doesn't mean you'd be required to upgrade, this could/should be a community effort given if you guys support or endorse that at all. But for that part to work out or happen, it might be necessary for more people to maintain this project? This is just a stab into the dark, but I feel like it's only @ifox handling everything? Surely doesn't work out since @ifox has other things to do as well and it cannot be expected from him to be available nor review everything right away.

Please excuse if this may come over harsh, it's just that I feel like there is a gap between the public facing side of twill.io which attracts people and businesses, but on the other side feels like this project is rather a burden that get's maintained on a low frequent level.

So tldr: If a business were to use twill, what can they expect? Would a rough estimation be possible for support of the project itself? How can others help? Are you looking for more maintainers or could that help with the longevity of the project?

I am currently not actively using this project myself, but our company decided to use it which made me do this post. Eventually I'll get to contribute actively to this codebase and I am willing to take on some tasks.

Thanks in advance.

**Comment by ersin-demirtas** [comment]
@lostdesign there is already a PR #740  for laravel 8 support, Twill 2.2 will have support for Laravel 8 https://github.com/area17/twill/pull/740.

**Comment by lostdesign** [comment]
Thanks, I am actively following the updates of the project across here and spectrum. The PR that is just answering half of what I questioned and leaves still quite something uncertain. 

I simply want to know what "twill" will be in the feature - since currently twill somewhat moves with the Laravel release cycle but kinda is behind and the maintenance of the project doesn't seem guaranteed or actively done. 

Laravel v6 would be LTS, perhaps it would make sense to keep twills base on the latest LTS to not always have the peer pressure upgrading twill every year. This gives the maintainers room to breath and the users security updates throughout the use until the next LTS is released. A security vulnerability that'd be discovered in march, wouldn't be fixed in many twill instances unless the maintainers would do so. If twill would use an LTS version, users could simply pull the latest Laravel version.

Don't get me wrong, that's just a random example. There simply isn't any clarity with twill imho - but the website kinda advertises it as _the_ CMS. I am not expecting or demanding Wordpress grade maintenance or support, it's simply enough to honestly tell the users what to expect.

I.e. twill is a hobbyist cms which is only recommended to people knowing laravel from the insides, those that are ready to fork and maintain their own version or if twill is planning on maintaining this project / looking for more maintainers..

**Comment by ghost** [comment]
I fully agree with the author above. I would like to clarify the update date to Laravel version 8

**Comment by CodeDredd** [comment]
@ersin-demirtas Thanks what are going to support Laravel 8.
I totally agree with @lostdesign . More over i am still worried about your Laravel 8 upgrade because you still struggle with https://github.com/Azure/azure-storage-php/issues/248 and i have seen no updates so far.

Don't get me wrong, i still like the work done in that CMS. But it seems more & more that this CMS is more for personal projects rather than commercial/enterprise ones.

**Comment by dvcom69** [comment]
Any news?

**Comment by zherdev-artem** [comment]
Any news?

**Comment by domihagen** [comment]
Is there any Update? When will be the next Twill release?

**Comment by mariusghitoiu** [comment]
Any news on this one? Laravel 7 is EOL today.

**Comment by DanielTate** [comment]
Maybe Twill is dead?

**Comment by lostdesign** [comment]
Sure is dead @DanielTate we've tried to contact them in multiple ways, haven't gotten any response - even offered to financially back this project in order to keep it alive. 🤷

**Comment by adam-code-labx** [comment]
Certainly not dead!
I assume they are hard at work with the next release. I receive daily updates when watching v2 branch. Why not use their 2.x branch which is compatible with Laravel 8 😉https://github.com/area17/twill/tree/2.x

**Comment by lostdesign** [comment]
Why not communicate what's going on as it's constantly being asked? Why turn down business enquiries? Why turn down offered help? The communication surely is dead. 😉

**Comment by adam-code-labx** [comment]
Can’t really answer that one as I am not part of the team, just pointing out there is an active branch which seems to be updated daily. IMO would be nice for them to open up a discord channel. Open source communication could certainly be improved Spectrum is out of date now, so even if Twill doesn’t have the time to update everyone at least we will all be able to chat as a community on discord. Just a suggestion 😜

**Comment by lostdesign** [comment]
Been preaching and actively offering help to take the community part as lead, at least in getting a server started and creating a place for the community. Done that before multiple times, currently bootstrapping the server for tiptap and running a rather successful server with 28k members as of now.

All I and or my company asks for:
- place for the community
- roadmap
- respond to business inquiries?

**Comment by lostdesign** [comment]
Now that v2.2 has been released, is it a sign of "our roadmap is to try to keep up with every new laravel version" or finally switch to LTS on the laravel v9 release later that year in september?

**Comment by vanvanni** [comment]
I think Twill can get big adoption if they have a better community support, more often updates to fit Laravel version and something like Discord. Also the docs are very in depth. But straight forward docs also isn't there. It's al very technical and detailed for the stuf you won't need 65% of the time.

**Reply by lostdesign** [reply]
I actually gotta defend twill at this point - kinda. There is a discord now and the support there is massive. They just kinda butcher communicating things. The docs part is correct however.
